Dom with javac mine

robstryker opened the pull request:
Several commits that add functionality and fix some test cases. 
Added two test cases for recent issues Fred found. 
Make small changes to the test suite to allow this and isolate our tests a bit.

mickaelistria commented on org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterBugsTest.java:
```diff
@@ -64,81 +61,6 @@ public static Test suite() {
 	return buildModelTestSuite(ASTConverterBugsTest.class);
 }
 
-protected void checkParameterAnnotations(String message, String expected, IMethodBinding methodBinding) {
```
As we'd rather avoid diverging with upstream JDT at that point, I think that -as bad is it is in general- it's better to just duplicate it to the new test class.

mickaelistria commented on org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/JavacASTConverterBugsTestJLS.java:
```diff
@@ -0,0 +1,101 @@
+ * as well.
+ */
+@SuppressWarnings("rawtypes")
+public class JavacASTConverterBugsTestJLS extends ASTConverterBugsTestSetup {
```
Should this Javac-specific test get into a javac-specific bundle?

mickaelistria commented:
I'm going to merge the changes in the CompilationUnitResolver and the Converter. We will keep this PR open to discuss the best strategy for testing.

mickaelistria commented:
The jdtCompilerAdapter.jar file should be excluded. If you can remove it soon, that's good; if not I'll tweak the commit tomorrow before pushing it.

Commit:
Several fixes to test cases

    Fix SortCompilationUnitElementsTests.test012

    Regression

    Fix SortCompilationUnitElementsTests.test012

    Regression

    Fix testAnnotationAST1

    Maybe fixes to #141, worth a shot!

    Add a test case for fred with super method invocations

    Fix compilation error

Signed-off-by: Rob Stryker <stryker@redhat.com>
